sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,331 @@
+            {
+                this.semanticModel = semanticModel;
+            }
+
+            public Expression BuildInstruction(SyntaxNode syntaxNode, BasicBlock block)
+            {
+                syntaxNode = syntaxNode.RemoveParentheses();
+
+                Expression expression;
+                if (nodeExpressionMap.TryGetValue(syntaxNode, out expression))
+                {
+                    return expression;
+                }
+
+                switch (syntaxNode.Kind())
+                {
+                    case SyntaxKind.AddExpression:
+                        expression = BuildBinaryExpression((BinaryExpressionSyntax)syntaxNode, block);
+                        break;
+
+                    case SyntaxKind.SimpleAssignmentExpression:
+                        expression = BuildAssignment((AssignmentExpressionSyntax)syntaxNode, block);
+                        break;
+
+                    case SyntaxKind.InvocationExpression:
+                        expression = BuildInvocation((InvocationExpressionSyntax)syntaxNode, block);
+                        break;
+
+                    case SyntaxKind.IdentifierName:
+                        expression = BuildIdentifier((IdentifierNameSyntax)syntaxNode, block);
+                        break;
+
+                    case SyntaxKind.VariableDeclarator:
+                        BuildVariableDeclarator((VariableDeclaratorSyntax)syntaxNode, block);
+                        expression = null;
+                        break;
+
+                    case SyntaxKind.ReturnStatement:
+                        BuildReturn((ReturnStatementSyntax)syntaxNode, block);
+                        expression = null;
+                        break;
+
+                    default:
+                        // do nothing
+                        expression = CreateConstant();
+                        break;
+                }
+
+                nodeExpressionMap.Add(syntaxNode, expression);
+                return expression;
+            }
+
+            private Expression BuildIdentifier(IdentifierNameSyntax identifier, BasicBlock block)
+            {
+                var identifierSymbol = GetSymbol(identifier);
+                if (identifierSymbol is IPropertySymbol property)
+                {
+                    var tempVariable = GetTempVariable();
+                    block.Instructions.Add(new Instruction
+                    {
+                        MethodId = MethodIdProvider.Create(property.GetMethod),
+                        Variable = tempVariable,
+                    });
+                    return CreateVariable(tempVariable);
+                }
+                else
+                {
+                    return CreateExpression(identifier);
+                }
+            }
+
+            private void BuildVariableDeclarator(VariableDeclaratorSyntax variableDeclarator, BasicBlock block)
+            {
+                if (variableDeclarator.Initializer == null)
```

Comment:
This should never be false, should it? An assertion would do.

Reply:
The initializer is null when `int x;`. When `int x = 1` the initializer is not null.